server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -506,20 +477,7 @@ public boolean removeSegment(String dataSourceName, final String segmentId)
   public boolean removeSegment(SegmentId segmentId)
```

Comment:
This method now looks the same as `public boolean removeSegment(String dataSourceName, final String segmentId)`, i. e. you can simplify (at least `dataSourceName` param is completely unused), potentially remove one of these methods completely. Also please update their Javadocs.

Reply:
both these methods are used from `MetadataSegmentManager`, so not sure if I can remove it. I made the `removeSegment(String dataSourceName, final String segmentId)` call `removeSegment(SegmentId segmentId)` internally. The javadoc for these is in the interface `MetadataSegmentManager`, will add a note here that they do not update the snapshot now.